scikit/notebooks/Scikit Learn on CMLE.ipynb

```diff
@@ -0,0 +1,381 @@
+    "# Create our training labels list\n",
+    "train_labels = (raw_training_data['income-level'] == ' >50K')\n",
+    "\n",
+    "\n",
+    "# Load the test census dataset\n",
+    "with tf.gfile.Open('./census_data/adult.test', 'r') as test_data:\n",
+    "    raw_testing_data = pd.read_csv(test_data, names=COLUMNS, skiprows=1)\n",
+    "# Remove the column we are trying to predict ('income-level') from our features list\n",
+    "test_features = raw_testing_data.drop('income-level', axis=1)\n",
+    "# Create our training labels list\n",
+    "test_labels = (raw_testing_data['income-level'] == ' >50K.')\n",
+    "\n",
+    "\n",
+    "# Convert the categorical columns to a numerical value in both the training and testing dataset\n",
+    "from sklearn.preprocessing import LabelEncoder\n",
+    "encoders = {col:LabelEncoder() for col in CATEGORICAL_COLUMNS}\n",
+    "\n",
+    "for col in CATEGORICAL_COLUMNS:\n",
+    "    train_features[col] = encoders[col].fit_transform(train_features[col])\n",
+    "\n",
+    "for col in CATEGORICAL_COLUMNS:\n",
+    "    test_features[col] = encoders[col].fit_transform(test_features[col])\n",
+    "\n",
+    "\n",
+    "# Create and train a classifier\n",
+    "classifier = RandomForestClassifier()\n",
+    "classifier.fit(train_features, train_labels)\n",
+    "\n",
+    "# Export the classifier to a file\n",
+    "joblib.dump(classifier, 'model.joblib')\n",
+    "\n",
+    "print('Done')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Part 2: Upload the model\n",
+    "Next, you'll need to upload the model to your project's storage bucket in GCS.\n",
+    "\n",
+    "**Replace** `BUCKET_ID` with the bucket id you used in the setup step.\n",
+    "\n",
+    "Note: The exact file name of of the exported model you upload to GCS is important! Your model must be named  “model.joblib”, “model.pkl”, or “model.bst” with respect to the library you used to export it. This restriction ensures that the model will be safely reconstructed later by using the same technique for import as was used during export."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "! gsutil cp ./model.joblib gs://[BUCKET_ID]/model.joblib"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Part 3: Create a Model resource\n",
+    "** Replace ** `MODEL_NAME` with the name for your model. (We used \"scikit_learn_prediction\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "! gcloud ml-engine models create [MODEL_NAME] --regions us-central1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Part 4: Create a Version resource\n",
+    "\n",
+    "** Replace: **\n",
```

Comment:
Could we actually define some defaults in code and allow users to change them if they would like? I would prefer if the whole notebook could be executed without anyone having to change any values.

`PROJECT` could perhaps be populated with `$(gcloud config get-value project)`? Everything else can be specified based on the project value.

Reply:
In general, if something does have to be specified by the user, I prefer to put it at the top of the notebook and call their attention to it right away. After that, they can "Execute all cells" and move on with their lives if they like.

Reply:
Fixed: User only has to do it at the beginning now.